### PR TITLE
using pydantic models in ogcapi.processes

### PR DESCRIPTION
**Pull request by cehbrecht**
this PR updates `ogcapi.processes` to use pydantic models (from #755):

* models are incomplete
* got it working with pygeoapi demo
* removed 52n demo

### CODE REVIEW

**Comment by cehbrecht** [comment]
models look useful for validation ... but currently difficult ... each demo service response different.

Providing outputs with models looks fine but very similar to json.

File: owslib/ogcapi/models.py [hunk 1]
```diff
@@ -129,7 +130,7 @@ class LiteralDataType(BaseModel):
 
 
 class ComplexDataType(BaseModel):
-    formats: List[FormatDescription]
+    formats: Optional[List[FormatDescription]]
```
Comment (huard) [inline owslib/ogcapi/models.py:133]: Do you need to add `= None` ?

File: owslib/ogcapi/models.py [hunk 2]
```diff
@@ -271,9 +272,20 @@ class Subscriber(BaseModel):
...
+
+class LiteralOutput(BaseModel):
+    id: str
+    value: str
+
+
+class Result(BaseModel):
+    jobID: Optional[str]
+    location: Optional[str] = None
+    outputs: Optional[List[LiteralOutput]] = None
```
Comment (huard) [inline owslib/ogcapi/models.py:291]: This does not match the schema.

Comment (huard) [reply owslib/ogcapi/models.py:291]: I think what is missing is JobList

**Review by huard — APPROVED** [review]

File: owslib/ogcapi/models.py (hunk 1 above)
Comment (cehbrecht) [reply owslib/ogcapi/models.py]: done

File: owslib/ogcapi/models.py (hunk 2 above)
Comment (cehbrecht) [reply owslib/ogcapi/models.py:308]: done.

**Pull request merged**